### PR TITLE
feat(storage): support delete source objects on compose

### PR DESCRIPTION
**Pull request by nidhiii-27**
Updates compose file sample to support deleting source objects optionally. Fixes b/441557254

### CODE REVIEW

File: google-cloud-storage/samples/acceptance/files_test.rb [hunk 1]
```diff
@@ -431,6 +431,26 @@ def mock_cipher.random_key
     refute_nil bucket.file remote_file_name
   end
 
+  it "compose_file with delete_source_objects" do
```
Comment (torreypayne) [inline google-cloud-storage/samples/acceptance/files_test.rb:437]: Nit: For resiliency against edge cases, it would be awesome to add a test to cover how the client behaves if a compose operation fails midway (ex. due to a transient API error). Otherwise LGTM!

File: google-cloud-storage/samples/acceptance/files_test.rb [hunk 2]
```diff
@@ -431,6 +431,26 @@ def mock_cipher.random_key
     refute_nil bucket.file remote_file_name
```
Comment (torreypayne) [inline google-cloud-storage/samples/acceptance/files_test.rb:431]: Nit: What do you think about asserting that the source files remain intact in this default test block above? I looked at the previous PR that adds in the `delete_source_objects` param and I'm wondering if we can add an extra check to validate the `delete_source_objects: false` value doesn't accidentally delete the source files.

**Review by torreypayne — APPROVED** [review]

File: google-cloud-storage/samples/acceptance/files_test.rb (hunk 2 above)
Comment (nidhiii-27) [reply google-cloud-storage/samples/acceptance/files_test.rb:431]: Done

_Co-authored by AI Agent_

File: google-cloud-storage/samples/acceptance/files_test.rb (hunk 1 above)
Comment (nidhiii-27) [reply google-cloud-storage/samples/acceptance/files_test.rb:437]: Done

_Co-authored by AI Agent_

File: google-cloud-storage/samples/acceptance/files_test.rb [hunk 3]
```diff
@@ -429,6 +429,52 @@ def mock_cipher.random_key
     end
 
     refute_nil bucket.file remote_file_name
+    refute_nil bucket.file file_1_name
```
Comment (cpriti-os) [inline google-cloud-storage/samples/acceptance/files_test.rb:433]: Add a comment to explain the assertion here.

File: google-cloud-storage/samples/acceptance/files_test.rb [hunk 4]
```diff
@@ -429,6 +429,52 @@ def mock_cipher.random_key
...
+                   destination_file_name: remote_file_name,
+                   delete_source_objects: true
+    end
+
+    refute_nil bucket.file remote_file_name
+    assert_nil bucket.file file_1_name
+    assert_nil bucket.file file_2_name
+  end
+
+  it "compose_file with delete_source_objects failing does not delete source objects" do
```
Comment (cpriti-os) [inline google-cloud-storage/samples/acceptance/files_test.rb]: This looks like a compose failure mock and not delete source object failure? Could you name it accordingly

File: google-cloud-storage/samples/acceptance/files_test.rb (hunk 3 above)
Comment (nidhiii-27) [reply google-cloud-storage/samples/acceptance/files_test.rb:433]: Done

_Co-authored by AI Agent_

File: google-cloud-storage/samples/acceptance/files_test.rb (hunk 4 above)
Comment (nidhiii-27) [reply google-cloud-storage/samples/acceptance/files_test.rb]: Done

_Co-authored by AI Agent_

**Review by cpriti-os — APPROVED** [review]
